salt/modules/glusterfs.py

```diff
@@ -43,34 +43,46 @@ def _get_minor_version():
     return version
 
 
-def _gluster(cmd):
+def _gluster_ok(xml_data):
     '''
-    Perform a gluster command.
+    Extract boolean return value from Gluster's XML output.
     '''
-    # We will pass the command string as stdin to allow for much longer
-    # command strings. This is especially useful for creating large volumes
-    # where the list of bricks exceeds 128 characters.
-    return __salt__['cmd.run'](
-        'gluster --mode=script', stdin="{0}\n".format(cmd))
+    return int(xml_data.find('opRet').text) == 0
 
 
 def _gluster_xml(cmd):
     '''
-    Perform a gluster --xml command and check for and raise errors.
+    Perform a gluster --xml command and log result.
     '''
+    # We will pass the command string as stdin to allow for much longer
+    # command strings. This is especially useful for creating large volumes
+    # where the list of bricks exceeds 128 characters.
     root = ET.fromstring(
         __salt__['cmd.run'](
             'gluster --xml --mode=script', stdin="{0}\n".format(cmd)
         ).replace("\n", ""))
-    if int(root.find('opRet').text) != 0:
-        raise CommandExecutionError(root.find('opErrstr').text)
```

Comment:
Notice, in my most recent PR, that I'm no longer using the string output for determining results.